torch/fx/experimental/proxy_tensor.py

```diff
@@ -158,6 +119,9 @@ def proxy_call(proxy_mode, func_overload, args, kwargs=None):
         if r is not NotImplemented:
             return r
 
+    # DODGY af
+    g = DisableTracingGuard(proxy_mode.sym_mode)
+
     all_constant = pytree.tree_all_only(ProxyTensor, lambda t: t.constant is not None, (args, kwargs))
```

Comment:
hmmmmm I'm still reading, but do we need to do the same check for constants on the `SymInt` arguments here?

Maybe it doesn't really matter, since today none of the native functions labeled with `data_dependent_output` in the yaml take in any `SymInt` arguments.

Reply:
yes, I'm actually working on this right now. Unfortunately this is difficult to test because there are other bugs right now. But here is the test that should raise an error for data dependent access which fails for other reasons right now

```
    def test_symbolic_constant(self):
        def f(x):
            y = torch.tensor([1.0])
            print(x.shape[0])
            z = y.new_empty(x.shape)
            z.fill_(0)
            return torch.sum(z).item()

        g = make_fx(f, tracing_mode="symbolic")(torch.randn(8))
        print(g)
        # g = make_fx(f, tracing_mode="symbolic")(torch.randn(2))
        # print(g)
```